JsonPatch.Tests/PatchExtensionTests.cs

```diff
@@ -0,0 +1,258 @@
+			Assert.AreNotEqual(initial.RootElement.ToString(),expected.RootElement.ToString());
+			var actual = patch.ApplyPatch(initial);
+			Assert.AreEqual(actual.RootElement.ToString(),expected.RootElement.ToString());
+		}
+
+		[Test]
+		public void RemoveArray_Test()
+		{
+			var initial = JsonDocument.Parse("[1,2,3]");
+			var expected = JsonDocument.Parse("[1,2]");
+			var patchExpected = "[{\"op\":\"remove\",\"path\":\"/2\"}]";
+			var patch = initial.CreatePatch(expected, new JsonSerializerOptions() {IgnoreNullValues = true});
+			
+			Assert.AreEqual(patch.Operations.Count, 1);
+			Assert.AreEqual(patch.Operations[0].Op, OperationType.Remove);
+			Assert.AreEqual(patchExpected, JsonSerializer.Serialize(patch));
+
+			Assert.AreNotEqual(initial.RootElement.ToString(),expected.RootElement.ToString());
+			initial = patch.ApplyPatch(initial);
+			Assert.AreEqual(initial.RootElement.ToString(),expected.RootElement.ToString());
+		}
+
+		[Test]
+		public void ReplaceArray_Test()
+		{
+			var initial = JsonDocument.Parse("[1,2,3]");
+			var expected = JsonDocument.Parse("[1,2,1]");
+			var patch = initial.CreatePatch(expected, new JsonSerializerOptions() {IgnoreNullValues = true});
+			var patchExpected = "[{\"op\":\"replace\",\"path\":\"/2\",\"value\":1}]";
+			
+			Assert.AreEqual(patch.Operations.Count, 1);
+			Assert.AreEqual(patch.Operations[0].Op, OperationType.Replace);
+			Assert.AreEqual(patchExpected, JsonSerializer.Serialize(patch));
+
+			Assert.AreNotEqual(initial.RootElement.ToString(),expected.RootElement.ToString());
+			initial = patch.ApplyPatch(initial);
+			Assert.AreEqual(initial.RootElement.ToString(),expected.RootElement.ToString());
+		}
+
+
+		[Test]
+		public void ComplexObject_Test()
+		{
+			var initial = new TestModel()
+			{
+				Id = Guid.Parse("aa7daced-c9fa-489b-9bc1-540b21d277a1"),
+				Attributes = JsonDocument.Parse("[{\"test\":\"test123\"},{\"test\":\"test32132\"},{\"test1\":\"test321\"},{\"test\":[1,2,3]},{\"test\":[1,2,3]}]"),
+				Name = "Test",
+				Numbers = new[] {1, 2, 3},
+				Strings = new[] {"test1", "test2"},
+				InnerObjects = new List<TestModel>() {new TestModel() {Id = Guid.Parse("b2cab2a0-ec23-405a-a5a8-975448a10334"), Name = "TestNameInner1", Numbers = new[] {3, 2, 1}, Strings = new[] {"Test3", "test4"}}}
+			};
+			var initialJson = JsonSerializer.Serialize(initial);
+			var expected = new TestModel()
+			{
+				Id = Guid.Parse("4801bd62-a8ec-4ef2-ae3c-52b9f541625f"),
+				Attributes = JsonDocument.Parse("[{\"test1\":\"test123\"},{\"test\":\"test32132\"},{\"test1\":\"test321\"},{\"test\":[1,1,3]}]"),
+				Name = "Test4",
+				Numbers = new[] {1, 2, 3, 4},
+				Strings = new[] {"test2", "test2"},
+				InnerObjects = new List<TestModel>() {new TestModel() {Id = Guid.Parse("bed584b0-7ccc-4336-adba-d0d7f7c3c3f2"), Name = "TestNameInner1", Numbers = new[] {1, 2, 1}, Strings = new[] {"Test3", "test4", "test5"}}}
+			};
+			var expectedJson = JsonSerializer.Serialize(expected);
+			var patchExpected =
```

Comment:
Yeah, I don't expect to check the patch in the complex tests; just the simple generation test.